homeassistant/config_entries.py

```diff
@@ -459,7 +459,7 @@ async def setup_again(*_: Any) -> None:
 
             await self._async_process_on_unload()
             return
-        except Exception:  # pylint: disable=broad-except
+        except (asyncio.CancelledError, Exception):  # pylint: disable=broad-except
```

Comment:
So, if the goal is to prevent integration to kill HA, there are more things to catch.

For example, an integration could now also raise an `SystemExit`, which would kill HA.

Shouldn't this just catch all? And thus have used `BaseException` instead of `Exception` in the first place?

```suggestion
        except (BaseException):  # pylint: disable=broad-except
```

Reply:
If we call SystemExit and KeyboardInterruptError, we should probably handle it and call `hass.async_stop` ?